pygeoapi/provider/hateoas.py

```diff
@@ -0,0 +1,205 @@
+                try:
+                    filename = os.path.basename(data_path)
+                    jsondata = _get_json_data('{}/{}.json'.format(data_path, filename))
+                    resource_type = 'Assets'
+                except Exception as err:
+                    msg = 'Resource does not exist: {}'.format(data_path)
+                    LOGGER.error(msg)
+                    raise ProviderNotFoundError(msg)
+
+
+        if resource_type == 'Catalog' or resource_type == 'Collection':
+            content['type'] = resource_type
+
+            link_href_list = []
+            for link in jsondata["links"]:
+                if (resource_type == 'Catalog' and link["rel"] in ["child", "item"]) or \
+                   (resource_type == 'Collection' and link["rel"] in ["child", "item"]):
+                    link_href_list.append(link["href"].replace('\\', '/'))
+            link_href_list.sort()
+
+            for link in link_href_list:
+                unused, path_ending, entry_type = link.split('/')
+                newpath = os.path.join(baseurl, urlpath, path_ending).replace('\\', '/')
+
+                if entry_type == 'catalog.json':
+                    child_links.append({
+                        'rel': 'child',
+                        'href': newpath,
+                        'type': 'text/html',
+                        'created': "-",
+                        'entry:type': 'Catalog'
+                    })
+                elif entry_type == 'collection.json':
+                    child_links.append({
+                        'rel': 'child',
+                        'href': newpath,
+                        'type': 'text/html',
+                        'created': "-",
+                        'entry:type': 'Collection'
+                    })
+                else:
+                    child_links.append({
+                        'rel': 'item',
+                        'href': newpath,
+                        'title': path_ending,
+                        'created': "-",
+                        'entry:type': 'Item'
+                    })
+
+        elif resource_type == 'Assets':
+            content = jsondata
+            content['assets']['default'] = {
+                'href': os.path.join(baseurl, urlpath).replace('\\', '/'),
+            }
+
+            for key in content['assets']:
+                content['assets'][key]['file:size'] = 0
+                content['assets'][key]['created'] = jsondata["properties"]["datetime"]
+
+        content['links'].extend(child_links)
+
+        return content
+
+    def __repr__(self):
+        return '<HateoasProvider> {}'.format(self.data)
+
+def _get_json_data(jsonpath):
+    """
+    Helper function used to load a json file that is located on the WEB (HTTP request)
+    or on the server file system
+
+    :param jsonpath: path to the json file
+
+    :returns: `dict` of JSON item
+    """
+
+    if jsonpath[0:4].upper() == 'HTTP':
```

Comment:
Do we need to account for HTTPS here i.e. options:

- `if jsonpath[0:4].upper().startswith('HTTP'):`
- `if jsonpath[0:4].upper() in ["HTTP', 'HTTPS']:`?

Reply:
The code, as it is, accepts both protocols since jsonpath[0:4].upper() returns the first four characters in uppercase and the first four characters of HTTPS are equal to HTTP. That said, option 1 is equivalent to "if jsonpath[0:4].upper() == 'HTTP':". However, it processes the entire path which is not necessary. Although option 2 is also equivalent to my code, it seems a bit strange because jsonpath[0:4].upper() returns the first 4 characters and HTTPS has 5 characters.